### PR TITLE
SWARM-772 remove Weld 3 modules.

### PR DESCRIPTION
**Pull request by tremes**
- [x ] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [ x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [ x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by tremes** [comment]
Needs new fraction plugin version - https://github.com/wildfly-swarm/wildfly-swarm-fraction-plugin/pull/24

File: core/container/src/main/resources/modules/javax/enterprise/api/main/module.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <module xmlns="urn:jboss:module:1.5" name="javax.enterprise.api">
```
Comment (kenfinnigan) [inline core/container/src/main/resources/modules/javax/enterprise/api/main/module.xml]: Couldn't this be completely removed in favor of the one from WildFly?

File: core/container/src/main/resources/modules/org/wildfly/swarm/container/runtime/module.xml [hunk 2]
```diff
@@ -49,10 +49,7 @@
     <module name="org.jboss.as.controller" export="true"/>
     <module name="org.jboss.as.self-contained"/>
     <module name="org.jboss.shrinkwrap"/>
-    <module name="org.jboss.weld.api" slot="3" export="true"/>
-    <module name="org.jboss.weld.spi" slot="3" export="true"/>
-    <module name="org.jboss.weld.core" slot="3"/>
-    <module name="org.jboss.weld.se" slot="3"/>
+    <module name="org.jboss.weld.se"/>
```
Comment (kenfinnigan) [inline core/container/src/main/resources/modules/org/wildfly/swarm/container/runtime/module.xml:52]: Why does this only get replaced with `org.jboss.weld.se` but the next file gets the modules replaced like for like?

File: fractions/wildfly/infinispan/module.conf [hunk 3]
```diff
@@ -1,7 +1,7 @@
 org.jboss.as.clustering.infinispan
 
 javax.enterprise.api
-org.jboss.weld.api:3
-org.jboss.weld.spi:3
-org.jboss.weld.core:3
-org.jboss.weld.se:3
+org.jboss.weld.api:main
```
Comment (kenfinnigan) [inline fractions/wildfly/infinispan/module.conf]: These don't need the `:main` suffix, as it's the default

File: pom.xml [hunk 4]
```diff
@@ -29,10 +29,9 @@
   </scm>
 
   <properties>
-    <version.wildfly.swarm.fraction.plugin>44</version.wildfly.swarm.fraction.plugin>
+    <version.wildfly.swarm.fraction.plugin>45-SNAPSHOT</version.wildfly.swarm.fraction.plugin>
```
Comment (kenfinnigan) [inline pom.xml:32]: Once fraction-plugin is updated and released this needs to be updated.

File: core/container/pom.xml [hunk 5]
```diff
@@ -25,9 +25,9 @@
   <properties>
     <swarm.fraction.internal>true</swarm.fraction.internal>
     <swarm.fraction.stability>stable</swarm.fraction.stability>
-    <version.weld-api>3.0.Alpha19</version.weld-api>
-    <version.weld>3.0.0.Alpha16</version.weld>
-    <version.cdi2>2.0.Alpha4</version.cdi2>
+    <version.weld-api>2.3.SP2</version.weld-api>
+    <version.weld>2.3.5.Final</version.weld>
+    <version.cdi>1.2</version.cdi>
```
Comment (kenfinnigan) [inline core/container/pom.xml:30]: Would not these artifacts be picked up from WildFly feature packs as is?

File: core/container/src/main/resources/modules/javax/enterprise/api/main/module.xml (hunk 1 above)
Comment (tremes) [reply core/container/src/main/resources/modules/javax/enterprise/api/main/module.xml]: Hopefully yes. Need to try out.

File: core/container/pom.xml (hunk 5 above)
Comment (tremes) [reply core/container/pom.xml:30]: I am not very familiar with WF feature packs. Doest it mean to be picked up from WF parent? Unfortunately there is no `weld-se-core` dependency in the WF parent so I guess we need to define it.

File: fractions/wildfly/infinispan/module.conf [hunk 6]
```diff
@@ -1,7 +1,7 @@
 org.jboss.as.clustering.infinispan
 
 javax.enterprise.api
-org.jboss.weld.api:3
-org.jboss.weld.spi:3
-org.jboss.weld.core:3
-org.jboss.weld.se:3
+org.jboss.weld.api
```
Comment (kenfinnigan) [inline fractions/wildfly/infinispan/module.conf]: I think these can be removed as the fraction-plugin adds the weld module for us

File: core/container/src/main/java/org/wildfly/swarm/container/runtime/cdi/CommonBean.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2015-2017 Red Hat, Inc, and individual contributors.
```
Comment (kenfinnigan) [inline core/container/src/main/java/org/wildfly/swarm/container/runtime/cdi/CommonBean.java]: Would this class and the Builder be better placed in the /spi?

Would enable fractions to utilize the bean creation pieces as well if they needed to

Comment (tremes) [reply core/container/src/main/java/org/wildfly/swarm/container/runtime/cdi/CommonBean.java]: Hm probably good idea.:-)

**Comment by kenfinnigan** [comment]
@tremes are you able to update and fix up this PR?

There are some conflicts with master branch

**Comment by tremes** [comment]
@kenfinnigan Rebased locally and there is only conflict in pom.xml wrt fraction plugin version. What's your plan to merge this? I am asking because the plugin needs to go as first so I would rebase this once there is new plugin version.

**Comment by kenfinnigan** [comment]
@bobmcwhirter has a separate change to fraction-plugin that would need to be merged and released before we could merge and release it again with your pr.

**Pull request merged**